Auth

BarretoJuan opened the pull request:
Auth Service for user log-in and sign-in

BarretoJuan committed:
First commit

alejandro0619 commented on app.py:
```diff
@@ -0,0 +1,100 @@
+        elif not re.match(r'[^@]+@[^@]+\.[^@]+', email):
+            msg = 'Dirección e-mail inválida'
+        #If form hasn't been completely filled
+        elif not email or not nombre or not password or not confirmarPassword:
```
Is there any reason why you would double check if the form is completely filled or not? I think the first `if` on line 59 already guarantee that every input in the form is already filled.

alejandro0619 commented on db/DBModel MindHive.mwb.bak:
This file shouldn't have been uploaded to github, add it to gitignore

alejandro0619 commented on __pycache__/app.cpython-311.pyc:
Should add this to gitignore

BarretoJuan committed:
Segundo commit

Correcciones en la logica del código

alejandro0619 commented on app.py:
```diff
@@ -91,9 +91,7 @@ def register():
             mysql.connection.commit()
             msg = 'Usuario registrado exitosamente'
             print(fechaCreacion)
-    elif request.method == 'POST':
```
As I understood this snippet would check if the method was POST but no field was filled.  If the `elif` is omitted, both POST and GET requests would fall into the  same condition of returning a view.

I'd keep this snippet as before.

alejandro0619 commented on app.py:
```diff
@@ -0,0 +1,98 @@
+
+    msg = ''
+    # Check if required fields exist in the form
+    if request.method == 'POST' and 'email' in request.form and 'nombre' in request.form and 'apellido' in request.form and "password" in request.form and "confirmarPassword" in request.form:
```
I would rewrite this snippet instead of the current for to avoid repetition:
```python
if request.method == 'POST' and all(field in request.form for field in ['email', 'nombre', 'apellido', 'password', 'confirmarPassword']):
# Code goes here
else:
# Code if any of the fields are not present in the form
```
This snippet would allow us to check if all the fields in the list (`['email', 'nombre', 'apellido', 'password', 'confirmarPassword']`) are present as keys in the dictionary `request.form`.

What do you think?

BarretoJuan replied:
looks neat!

BarretoJuan committed:
Auth Commit

Returned register post method to the way it was before. Created a logout endpoint. Modified home page to be shown only when a user is logged in, and implemented a logout button. Added a condition to check if the user has already logged in for the login and register endpoints, if the user is logged, it will be redirected to the homePage endpoint. modified the password_hash field for the user table, this field was set as UNIQUE by mistake, this prevented different users to have the same password, as the hash generated would've been the same, breaking the UNIQUE condition

BarretoJuan committed:
suggested snippet

added a suggested modification to the register endpoint logic by alejandro0619

alejandro0619 commented:
Ready to merge!
